script.js

```diff
@@ -1,4 +1,542 @@
+      // Add Player 1 Score
+      player1Score += player1Num;
+      return `Player 1 rolled ${player1VarDice}<br>Player 1 number is ${player1Num}<br>Player 2 click submit to play`;
+    }
+    if (gameStage == 3) {
+      var diceCounter = 0;
+      var player2VarDice = [];
+      while (diceCounter < numOfDice) {
+        player2VarDice.push(generateRandomDiceNumber());
+        console.log(player2VarDice);
+        diceCounter += 1;
+      }
+      player2VarDice.sort((a, b) => b - a);
+      console.log(player2VarDice);
+      player2Num = Number(player2VarDice.join(``));
+      // swtich game stage
+      gameStage = 5;
+      // Add Player 2 Score
+      player2Score += player2Num;
+      return `Player 2 rolled ${player2VarDice}<br>Player 2 number is ${player2Num}<br>Player 2 click submit to find the winner.`;
+    }
+  } else if (gameMode == 6) {
+    if (gameStage == 1) {
+      if (Number.isNaN(Number(input)) == true || !input || input == 0) {
+        // check if input is a number
+        return gameInstruction3;
+      } else {
+        numOfDice = input;
+        // switch game stage
+        gameStage = 2;
+        return `Each Player will have ${numOfDice} dice, click submit to start playing`;
+      }
+    }
+    if (gameStage == 2) {
+      var diceCounter = 0;
+      var player1VarDice = [];
+      while (diceCounter < numOfDice) {
+        player1VarDice.push(generateRandomDiceNumber());
+        console.log(player1VarDice);
+        diceCounter += 1;
+      }
+      player1VarDice.sort((a, b) => a - b);
+      console.log(player1VarDice);
+      player1Num = Number(player1VarDice.join(``));
+      // swtich game stage
+      gameStage = 3;
+      // switch player
+      currentPlayer = player2;
+      // Add Player 1 Score
+      player1Score += player1Num;
+      return `Player 1 rolled ${player1VarDice}<br>Player 1 number is ${player1Num}<br>Player 2 click submit to play`;
+    }
+    if (gameStage == 3) {
+      var diceCounter = 0;
+      var player2VarDice = [];
+      while (diceCounter < numOfDice) {
+        player2VarDice.push(generateRandomDiceNumber());
+        console.log(player2VarDice);
+        diceCounter += 1;
+      }
+      player2VarDice.sort((a, b) => a - b);
+      console.log(player2VarDice);
+      player2Num = Number(player2VarDice.join(``));
+      // swtich game stage
+      gameStage = 5;
+      // Add Player 2 Score
+      player2Score += player2Num;
+      return `Player 2 rolled ${player2VarDice}<br>Player 2 number is ${player2Num}<br>Player 2 click submit to find the winner.`;
+    }
+  } else if (gameMode == 7) {
+    if (gameStage == 1) {
+      if (Number.isNaN(Number(input)) == true || !input || input == 1) {
+        // check if input is a number
+        return gameInstruction4;
+      } else {
+        numOfPlayer = input;
+        // switch game stage
+        gameStage = 2;
+        return `There will be ${numOfPlayer} players, click submit to start playing`;
+      }
+    }
```

Comment:
Likewise, it can be quite hard for someone to come in and decipher all this code since there are a lot of if and else ifs and nested ifs and else ifs. I think that naming the gameModes numerically can make it super difficult to decipher at first since they will need to keep referring to the list of gameModes while traversing through this code. I am not doubting that this code works but I think that many of the ifs and else ifs can be cut out

Also, we should keep the main function as short as possible - I think its best to segment out the game logic into its own helper functions. That way, people can see the helper function being called and see which gamemode they are currently reading about. It also helps to abstract the game logic away